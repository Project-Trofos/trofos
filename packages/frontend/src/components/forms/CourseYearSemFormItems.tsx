import React from 'react';
import { Form, Select } from 'antd';
import { DatePicker } from '../datetime';

type CourseYearSemFormItemsProps = {
  yearLabel?: string;
  semLabel?: string;
  yearName?: string;
  semName?: string;
};

// Renders form elements to input year and semester
export default function CourseYearSemFormItems({
  yearLabel = 'Academic Year',
  semLabel = 'Semester',
  yearName = 'courseYear',
  semName = 'courseSem',
}: CourseYearSemFormItemsProps): JSX.Element {
  return (
    <>
      <Form.Item
        label={yearLabel}
        name={yearName}
        rules={[{ required: true, message: "Please input your course's year!" }]}
      >
        <DatePicker picker="year" />
      </Form.Item>

      <Form.Item
        label={semLabel}
        name={semName}
        rules={[{ required: true, message: "Please input your course's semester!" }]}
      >
        <Select placeholder="Select a semester">
          <Select.Option key="1" value="1">
            1
          </Select.Option>
          <Select.Option key="2" value="2">
            2
          </Select.Option>
        </Select>
      </Form.Item>
    </>
  );
}
