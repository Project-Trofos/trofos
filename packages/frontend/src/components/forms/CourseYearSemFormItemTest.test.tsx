import React from 'react';
import { render, screen } from '@testing-library/react';
import FormWrapper from './__testutils__/FormRenderHelper';
import CourseYearSemFormItems from './CourseYearSemFormItems';

describe('CourseYearSemesterFormItem test', () => {
  function renderForm(yearName?: string, yearLabel?: string, semName?: string, semLabel?: string) {
    return render(
      <FormWrapper>
        <CourseYearSemFormItems yearName={yearName} yearLabel={yearLabel} semName={semName} semLabel={semLabel} />
      </FormWrapper>,
    );
  }

  describe('when rendering', () => {
    it('should have correct fields', () => {
      const { baseElement } = renderForm();
      expect(baseElement).toMatchSnapshot();

      expect(screen.getByLabelText(/academic year/i)).toBeInTheDocument();
      expect(screen.getByLabelText(/semester/i)).toBeInTheDocument();
    });

    it('should allow changing of label and name', () => {
      const YEAR_NAME = 'year';
      const YEAR_LABEL = YEAR_NAME;
      const SEM_NAME = 'sem';
      const SEM_LABEL = SEM_NAME;
      renderForm(YEAR_NAME, YEAR_LABEL, SEM_NAME, SEM_LABEL);

      expect(screen.getByLabelText(YEAR_LABEL)).toBeInTheDocument();
      expect(screen.getByLabelText(SEM_LABEL)).toBeInTheDocument();
    });
  });
});
