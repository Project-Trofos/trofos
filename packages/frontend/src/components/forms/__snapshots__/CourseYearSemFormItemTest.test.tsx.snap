// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseYearSemesterFormItem test when rendering should have correct fields 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="courseYear"
              title="Academic Year"
            >
              Academic Year
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-picker"
                >
                  <div
                    class="ant-picker-input"
                  >
                    <input
                      aria-required="true"
                      autocomplete="off"
                      id="courseYear"
                      placeholder="Select year"
                      readonly=""
                      size="12"
                      title=""
                      value=""
                    />
                    <span
                      class="ant-picker-suffix"
                    >
                      <span
                        aria-label="calendar"
                        class="anticon anticon-calendar"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="calendar"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="courseSem"
              title="Semester"
            >
              Semester
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="courseSem_list_0"
                        aria-autocomplete="list"
                        aria-controls="courseSem_list"
                        aria-haspopup="listbox"
                        aria-owns="courseSem_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="courseSem"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select a semester
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default"
        type="submit"
      >
        <span>
          submit
        </span>
      </button>
    </form>
  </div>
</body>
`;
